nselib/url.lua

```diff
@@ -177,8 +178,22 @@ function parse(url, default)
     parsed.params = p
     return ""
   end)
+
   -- path is whatever was left
   parsed.path = url
+
+  -- Checks for folder route and extension
+  if(string.match(parsed.path, "/$")) then
+    parsed.is_folder = true
+  else
+    parsed.is_folder = false
+    local split_str = stdnse.strsplit('%.', parsed.path)
+    if(split_str and #split_str > 1) then
+      parsed.extension = stdnse.strsplit("%;", split_str[#split_str])[1]
+    end
```

Comment:
`/files/tmp.php/lastfile.txt` - Really? Does this exist?
`/files/tmp.php/lastfile`
`/files.` - Why would anyone put `.` at the end? Even if someone puts it I think we can ignore these kind of extensions because its not a extension at all.

@nnposter  Do the above kind of urls exist in real world or was it just a hypothetical example? If they do exist could you give me a live url (I just didn't find any).